app/shared/controllers/validators/resolvers/ResolveDateRange.scala

```diff
@@ -16,60 +16,56 @@
 
 package shared.controllers.validators.resolvers
 
-import shared.models.domain.DateRange
-import shared.models.errors.{EndDateFormatError, MtdError, RuleEndBeforeStartDateError, StartDateFormatError}
 import cats.data.Validated
 import cats.data.Validated.{Invalid, Valid}
 import cats.implicits._
+import shared.models.domain.DateRange
+import shared.models.errors.{EndDateFormatError, MtdError, RuleEndBeforeStartDateError, StartDateFormatError}
 
 import java.time.LocalDate
+import scala.math.Ordering.Implicits.infixOrderingOps
 
-private[resolvers] class ResolveDateRange private (yearLimits: Option[YearLimits]) extends Resolver[(String, String), DateRange] {
+case class ResolveDateRange(startDateFormatError: MtdError = StartDateFormatError,
+                            endDateFormatError: MtdError = EndDateFormatError,
+                            endBeforeStartDateError: MtdError = RuleEndBeforeStartDateError)
+    extends ResolverSupport {
 
-  def apply(value: (String, String), notUsedError: Option[MtdError], path: Option[String]): Validated[Seq[MtdError], DateRange] = {
-    val (startDate, endDate) = value
+  private def resolveDateRange(parsedStartDate: LocalDate, parsedEndDate: LocalDate): Validated[Seq[MtdError], DateRange] =
+    if (parsedEndDate <= parsedStartDate)
+      Invalid(List(endBeforeStartDateError))
+    else
+      Valid(DateRange(parsedStartDate, parsedEndDate))
 
-    val resolvedDates = (
-      ResolveIsoDate(startDate, StartDateFormatError),
-      ResolveIsoDate(endDate, EndDateFormatError)
+  val resolver: Resolver[(String, String), DateRange] = { case (startDate, endDate) =>
```

Comment:
As with the other "val resolvers", I was wondering if these can be private, as they tend to be used only internally. Or are they also intended for "public" use?

Reply:
They are intended for public use. Especially when there is ad hoc or API-specific validation that is required to be applied to a common (& shared) 'resolver'. E.g. where an API or group of validator factories has particular needs, it doesn't seem worth creating a method in the shared resolver (that might not be used in other APIs). 

This pattern where validator factories use (`resolver thenValidate ...` or whatever we want to call it) could have been applied more widely but I wanted to keep callers unchanged where possible.